app/src/main/java/com/example/easyupipayment/MainActivity.java

```diff
@@ -27,64 +31,122 @@ protected void onCreate(Bundle savedInstanceState) {
-                .build();
-
-        //Register Listener for Events
-        easyUpiPayment.setPaymentStatusListener(this);
-
         //Proceed for Payment on click
         payButton.setOnClickListener(new View.OnClickListener() {
             @Override
             public void onClick(View v) {
-                easyUpiPayment.startPayment();
+                final String AMOUNT = amount.getText().toString();
+                final String UPI = upiid.getText().toString();
+                final String NOTE = note.getText().toString();
+                final String PayeeName = payee.getText().toString();
+
+                //Create instance of EasyUpiPayment
+                if(AMOUNT.isEmpty() || UPI.isEmpty() || NOTE.isEmpty() || PayeeName.isEmpty()){
+                    Toast.makeText(MainActivity.this, "All fields are mandatory.", Toast.LENGTH_SHORT).show();
+                }else {
+                    double amt = Double.parseDouble(AMOUNT);
+                    final EasyUpiPayment easyUpiPayment = new EasyUpiPayment.Builder()
+                            .with(MainActivity.this)
+                            .setPayeeVpa(UPI)
+                            .setDescription(NOTE)
+                            .setAmount(String.valueOf(amt))
+                            .setPayeeName(PayeeName)
+                            .setTransactionId("TRANSACTION_ID")
+                            .setTransactionRefId("TRANSACTION_REF_ID")
+                            .build();
+
+                    //Register Listener for Events
+                    easyUpiPayment.setPaymentStatusListener(MainActivity.this);
+
+                    easyUpiPayment.startPayment();
+                }
             }
         });
     }
 
+    String TranscationDetails;
+
     @Override
     public void onTransactionCompleted(TransactionDetails transactionDetails) {
         // Transaction Completed
         Log.d("TransactionDetails", transactionDetails.toString());
-        statusView.setText(transactionDetails.toString());
+        TranscationDetails =  transactionDetails.toString();
     }
 
     @Override
     public void onTransactionSuccess() {
         // Payment Success
-        Toast.makeText(this, "Success", Toast.LENGTH_SHORT).show();
-        imageView.setImageResource(R.drawable.ic_success);
+        new AlertDialog.Builder(MainActivity.this).setIcon(R.drawable.ic_success).setTitle("Transaction Successful")
+                .setMessage(TranscationDetails)
+                .setPositiveButton("Okay", new DialogInterface.OnClickListener() {
+                    @Override
+                    public void onClick(DialogInterface dialog, int which) {
+                        Intent intent = new Intent(Intent.ACTION_MAIN);
+                        intent.addCategory(Intent.CATEGORY_HOME);
+                        intent.setFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
+                        startActivity(intent);
+                        finish();
```

Comment:
Instead of this, to dismiss this dialog we can directly call `dialog.dismiss()` on `DialogInterface`